From 38f659a1d41d767cf917411dd0fe2c4526dac195 Mon Sep 17 00:00:00 2001
From: Grzegorz Jaszczyk <jaz@semihalf.com>
Date: Thu, 25 Apr 2019 18:06:41 +0200
Subject: [PATCH 194/386] arm64: dts: a8040-F: add new toplogy with internal
 eth mode

Add support for configuration F, which is the same as
"armada-8040-db.dts" but with internal phy-mode configuration
set for each CP PPv2 port0.

Change-Id: I6bd41f5828165d08dd482752ee181a061e2ada8d
Signed-off-by: Grzegorz Jaszczyk <jaz@semihalf.com>
Reviewed-on: https://sj1git1.cavium.com/8142
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Marcin Wojtas <marcin@marvell.com>
Reviewed-by: Stefan Chulski <Stefan.Chulski@cavium.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 .../bindings/arm/marvell/armada-7k-8k.txt     |  3 +++
 arch/arm64/boot/dts/marvell/Makefile          |  1 +
 .../boot/dts/marvell/armada-8040-db-F.dts     | 26 +++++++++++++++++++
 3 files changed, 30 insertions(+)
 create mode 100644 arch/arm64/boot/dts/marvell/armada-8040-db-F.dts

diff --git a/Documentation/devicetree/bindings/arm/marvell/armada-7k-8k.txt b/Documentation/devicetree/bindings/arm/marvell/armada-7k-8k.txt
index 668ad08674fa..6a1609d14ac9 100644
--- a/Documentation/devicetree/bindings/arm/marvell/armada-7k-8k.txt
+++ b/Documentation/devicetree/bindings/arm/marvell/armada-7k-8k.txt
@@ -57,6 +57,9 @@ Boards:
    - "marvell,armada8040-db-E", "marvell,armada8040-db"
       when the board being used is the Armada 8040 Development board with Setup E
 
+   - "marvell,armada8040-db-F", "marvell,armada8040-db"
+      when the board being used is the Armada 8040 Development board with Setup F
+
    - "marvell,armada8040-mcbin"
       when the board being used is the MACCHIATOBin community board
 
diff --git a/arch/arm64/boot/dts/marvell/Makefile b/arch/arm64/boot/dts/marvell/Makefile
index 2c582578cadf..aa6091e6e491 100644
--- a/arch/arm64/boot/dts/marvell/Makefile
+++ b/arch/arm64/boot/dts/marvell/Makefile
@@ -17,6 +17,7 @@ dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-db-B.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-db-C.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-db-D.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-db-E.dtb
+dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-db-F.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-db-G.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-db-H.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-mcbin.dtb
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db-F.dts b/arch/arm64/boot/dts/marvell/armada-8040-db-F.dts
new file mode 100644
index 000000000000..2fc797dbafe1
--- /dev/null
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db-F.dts
@@ -0,0 +1,26 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright (C) 2019 Marvell Technology Group Ltd.
+ *
+ * Device Tree file for Marvell Armada 8040 Development board
+ * This file supports option F configuration, which is the same as
+ * "armada-8040-db.dts" but with internal phy-mode configuration.
+ */
+
+#include <dt-bindings/gpio/gpio.h>
+#include "armada-8040-db.dts"
+
+/ {
+	model = "Marvell Armada 8040 development board default (F) setup";
+	compatible = "marvell,armada8040-db-F", "marvell,armada8040-db",
+		"marvell,armada8040", "marvell,armada-ap806-quad",
+		"marvell,armada-ap806";
+};
+
+&cp0_eth0 {
+	phy-mode = "internal";
+};
+
+&cp1_eth0 {
+	phy-mode = "internal";
+};
-- 
2.17.1

